presto-spi/src/main/java/io/prestosql/spi/type/DoubleType.java

```diff
@@ -88,6 +90,12 @@ public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int ri
         return Double.compare(leftValue, rightValue);
     }
 
+    @Override
+    public Optional<Range> getRange()
+    {
+        return Optional.of(new Range(-Double.MIN_VALUE, Double.MAX_VALUE));
```

Comment:
`-Double.MIN_VALUE` is some negative number very close to zero.
Probably `-Double.MAX_VALUE` is the right thing.

Please add a comment why the range does not include `POSITIVE_INFINITY`, `NEGATIVE_INFINITY`.

Reply:
Ah yes, I made that mistake with Float and fixed it, but forgot to update this one. I'll add a test, too.

Infinity and NaN are strange beasts, since they don't represent discrete values in the domain of the type. I'll probably have to add some special-casing for those values, anyway.

Reply:
> Ah yes, I made that mistake with Float and fixed it

:) this is why i noticed.
I don't use this constants frequently, so I wouldn't notice they are ill-named.

> Infinity and NaN are strange beasts, since they don't represent discrete values in the domain of the type.

Yup, they deserve a special treatment. 
You could think of Infinity as a special value that is just greater than other values. 
So the rule works because:
- infinity, nan cannot be produced by a cast from whole-integer type to floating point
- infinity, nan cannot be cast to whole-integer type

However, there is some issue still, i will add a comment inline.